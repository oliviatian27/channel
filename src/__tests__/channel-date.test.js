import React from 'react';
import { shallow,mount } from 'enzyme';
import moment from 'moment'

import ChannelDate from '../components/channel-list/channel-date';


describe('ChannelDate tests', () => {

  it('renders Channel-date', () => {
    const date = "2019-01-05";
    const wrapper = shallow(<ChannelDate date={date} />);

    // Expect the wrapper object to be defined
    expect(wrapper.find('.channel-date')).toBeDefined();
  });

  it('displays date based on props.date', () => {
    const date = "2019-01-01";
    const wrapper = mount(<ChannelDate date={date} />);

    // Expect the date text to be props.date
    expect(wrapper.find('time').at(0).text()).toBe(moment(date).format("ddd, MMMM DD, YYYY"));

    // update the date props
    wrapper.setProps({date:"2019-01-15"})

    expect(wrapper.find('time').at(0).text()).toBe(moment("2019-01-15").format("ddd, MMMM DD, YYYY"));

    wrapper.unmount()
  });
})
