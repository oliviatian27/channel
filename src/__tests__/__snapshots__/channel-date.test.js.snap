// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelDate tests renders Channel-date 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <ChannelDate
      date="2019-01-05"
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <t
          ago={false}
          calendar={false}
          decimal={false}
          element="time"
          filter={[Function]}
          format="ddd, MMMM DD, YYYY"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          2019-01-05
        </t>,
        "className": "channel-date",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ago": false,
          "calendar": false,
          "children": "2019-01-05",
          "decimal": false,
          "element": "time",
          "filter": [Function],
          "format": "ddd, MMMM DD, YYYY",
          "fromNow": false,
          "interval": 60000,
          "onChange": [Function],
          "titleFormat": "",
          "toNow": false,
          "unit": null,
          "unix": false,
          "utc": false,
          "withTitle": false,
        },
        "ref": null,
        "rendered": "2019-01-05",
        "type": [Function],
      },
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <t
            ago={false}
            calendar={false}
            decimal={false}
            element="time"
            filter={[Function]}
            format="ddd, MMMM DD, YYYY"
            fromNow={false}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            2019-01-05
          </t>,
          "className": "channel-date",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ago": false,
            "calendar": false,
            "children": "2019-01-05",
            "decimal": false,
            "element": "time",
            "filter": [Function],
            "format": "ddd, MMMM DD, YYYY",
            "fromNow": false,
            "interval": 60000,
            "onChange": [Function],
            "titleFormat": "",
            "toNow": false,
            "unit": null,
            "unix": false,
            "utc": false,
            "withTitle": false,
          },
          "ref": null,
          "rendered": "2019-01-05",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <t
        ago={false}
        calendar={false}
        decimal={false}
        element="time"
        filter={[Function]}
        format="ddd, MMMM DD, YYYY"
        fromNow={false}
        interval={60000}
        onChange={[Function]}
        titleFormat=""
        toNow={false}
        unit={null}
        unix={false}
        utc={false}
        withTitle={false}
      >
        2019-01-05
      </t>,
      "className": "channel-date",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ago": false,
        "calendar": false,
        "children": "2019-01-05",
        "decimal": false,
        "element": "time",
        "filter": [Function],
        "format": "ddd, MMMM DD, YYYY",
        "fromNow": false,
        "interval": 60000,
        "onChange": [Function],
        "titleFormat": "",
        "toNow": false,
        "unit": null,
        "unix": false,
        "utc": false,
        "withTitle": false,
      },
      "ref": null,
      "rendered": "2019-01-05",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <t
          ago={false}
          calendar={false}
          decimal={false}
          element="time"
          filter={[Function]}
          format="ddd, MMMM DD, YYYY"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          2019-01-05
        </t>,
        "className": "channel-date",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ago": false,
          "calendar": false,
          "children": "2019-01-05",
          "decimal": false,
          "element": "time",
          "filter": [Function],
          "format": "ddd, MMMM DD, YYYY",
          "fromNow": false,
          "interval": 60000,
          "onChange": [Function],
          "titleFormat": "",
          "toNow": false,
          "unit": null,
          "unix": false,
          "utc": false,
          "withTitle": false,
        },
        "ref": null,
        "rendered": "2019-01-05",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <t
          ago={false}
          calendar={false}
          decimal={false}
          element="time"
          filter={[Function]}
          format="ddd, MMMM DD, YYYY"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          2019-01-05
        </t>,
        "className": "channel-date",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ago": false,
          "calendar": false,
          "children": "2019-01-05",
          "decimal": false,
          "element": "time",
          "filter": [Function],
          "format": "ddd, MMMM DD, YYYY",
          "fromNow": false,
          "interval": 60000,
          "onChange": [Function],
          "titleFormat": "",
          "toNow": false,
          "unit": null,
          "unix": false,
          "utc": false,
          "withTitle": false,
        },
        "ref": null,
        "rendered": "2019-01-05",
        "type": [Function],
      },
      "type": "div",
    },
  ],
}
`;
